src/mc-fle2-payload-iev-v2.c

```diff
@@ -412,26 +422,92 @@ bool mc_FLE2IndexedRangeEncryptedValueV2_parse(mc_FLE2IndexedEncryptedValueV2_t
     CHECK_AND_RETURN(mc_reader_read_u8(&reader, &iev->bson_value_type, status));
 
     /* Read edge_count */
-    CHECK_AND_RETURN(mc_reader_read_u8(&reader, &iev->edge_count, status));
+    // Set equality edge_count to 1 as it doesn't technically exist but
+    // there will be a singular metadata block
+    if (iev->type == kTypeEquality) {
+        iev->edge_count = 1;
+    } else {
+        CHECK_AND_RETURN(mc_reader_read_u8(&reader, &iev->edge_count, status));
+    }
+
     // Maximum edge_count(255) times kMetadataLen(96) fits easily without
     // overflow.
-    const uint64_t edges_len = iev->edge_count * kMetadataLen;
+    const uint64_t metadata_len = iev->edge_count * kMetadataLen;
 
     /* Read ServerEncryptedValue. */
-    const uint64_t min_required_len = kMinServerEncryptedValueLen + edges_len;
-    const uint64_t SEV_and_edges_len = mc_reader_get_remaining_length(&reader);
-    if (SEV_and_edges_len < min_required_len) {
+    const uint64_t min_required_len = kMinServerEncryptedValueLen + metadata_len;
+    const uint64_t SEV_and_metadata_len = mc_reader_get_remaining_length(&reader);
+    if (SEV_and_metadata_len < min_required_len) {
         CLIENT_ERR("Invalid payload size %" PRIu64 ", smaller than minimum length %" PRIu64,
-                   SEV_and_edges_len,
+                   SEV_and_metadata_len,
                    min_required_len);
         return false;
     }
-    const uint64_t SEV_len = SEV_and_edges_len - edges_len;
+    const uint64_t SEV_len = SEV_and_metadata_len - metadata_len;
     CHECK_AND_RETURN(mc_reader_read_buffer(&reader, &iev->ServerEncryptedValue, SEV_len, status));
 
-    // Ignore Metadata block.
-    BSON_ASSERT(mc_reader_get_remaining_length(&reader) == edges_len);
+    iev->metadata = (mc_FLE2TagAndEncryptedMetadataBlock_t *)bson_malloc0(
+        iev->edge_count * sizeof(mc_FLE2TagAndEncryptedMetadataBlock_t));
+    if (!iev->metadata) {
+        CLIENT_ERR("Failed to allocate memory for metadata array");
+        return false;
+    }
```

Comment:
```suggestion
```

Remove unnecessary check. `bson_malloc0` aborts on [failure to allocate](https://mongoc.org/libbson/1.28.1/bson_malloc0.html).